.github/workflows/codeql-analysis.yml

```diff
@@ -7,8 +7,15 @@ on:
   schedule:
     - cron: '30 5,17 * * *'
 
+permissions:
+  contents: read
+
 jobs:
   analyze:
+    permissions:
+      actions: read  # for github/codeql-action/init to get workflow details
```

Comment:
`actions` and `contents` permissions are not needed for public repositories. https://github.com/github/codeql-action/blob/main/README.md#usage

Default restricted permission level for actions is none since our model is permissive do we need to define actions and contents permission.

Reply:
Thanks for reviewing the PR and flagging these unnecessary permissions. I have incorporated your feedback in ca261ac15c0a0b8e011701d5990ed829eea96307. To make sure that I understand the second part of your comment correctly, could you please clarify the ask with an example?

Reply:
Thanks alot @boahc077 and i am sorry about being not clear. What i was mentioned was our default GitHub token permission level is permissive so defining actions and contents permissions is not required. 
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token